### PR TITLE
ocm/rcm-controller: initial setup

### PR DESCRIPTION
**Pull request by cdoan1**

### CODE REVIEW

File: ci-operator/config/open-cluster-management/rcm-controller/open-cluster-management-rcm-controller-master.yaml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    namespace: open-cluster-management
+    tag: registration
+  registration-operator:
+    name: "4.5"
+    namespace: open-cluster-management
+    tag: registration-operator
+  registration-operator-src:
+    name: "4.5"
+    namespace: open-cluster-management
+    tag: registration-operator
```
Comment (pmorie) [inline ci-operator/config/open-cluster-management/rcm-controller/open-cluster-management-rcm-controller-master.yaml:17]: i think this needs to be `registration-operator-src`

Comment (cdoan1) [reply ci-operator/config/open-cluster-management/rcm-controller/open-cluster-management-rcm-controller-master.yaml]: updated

**Comment by cdoan1** [comment]
* adding the .config.prowgen fixed the clone problem
* now getting vbh setup error
* `image` step is getting 

```
Pulling image openshift/origin-release:golang-1.13 ...
Pulled 1/2 layers, 71% complete
Pulled 2/2 layers, 100% complete
Extracting
--> FROM openshift/origin-release:golang-1.13 as builder
--> ENV "foo"="bar"
--> ARG REMOTE_SOURCE
--> ARG REMOTE_SOURCE_DIR
--> ARG GITHUB_TOKEN
--> ARG COMPONENT_VERSION
--> ENV USE_VENDORIZED_BUILD_HARNESS=true
--> ENV COMPONENT_NAME=rcm-controller
--> ENV COMPONENT_VERSION="${COMPONENT_VERSION}"
--> ENV COMPONENT_TAG_EXTENSION=" "
--> COPY $REMOTE_SOURCE $REMOTE_SOURCE_DIR/app/
--> WORKDIR $REMOTE_SOURCE_DIR/app
--> RUN git config --global url."https://${GITHUB_TOKEN}:x-oauth-basic@github.com/".insteadOf "https://github.com/"
/bin/sh: line 0: cd: /app: No such file or directory
error: build error: running 'git config --global url."https://${GITHUB_TOKEN}:x-oauth-basic@github.com/".insteadOf "https://github.com/"' failed with exit code 1
2020/06/16 19:56:24 No custom metadata found and prow metadata already exists. Not updating the metadata.
2020/06/16 19:56:25 Ran for 1m20s
error: some steps failed:
  * could not run steps: step rcm-controller failed: could not wait for build: the build rcm-controller failed after 38s with reason DockerBuildFailed: Docker build strategy has failed.
--> COPY $REMOTE_SOURCE $REMOTE_SOURCE_DIR/app/
--> WORKDIR $REMOTE_SOURCE_DIR/app
--> RUN git config --global url."https://${GITHUB_TOKEN}:x-oauth-basic@github.com/".insteadOf "https://github.com/"
/bin/sh: line 0: cd: /app: No such file or directory
error: build error: running 'git config --global url."http.../".insteadOf "https://github.com/"' failed with exit code 1
2020/06/16 19:56:25 could not load result reporting options: failed to read file "": open : no such file or directory
```

**Comment by cdoan1** [comment]
after updating build command, 
now getting 

```
build github.com/open-cluster-management/rcm-controller/cmd/manager: cannot load github.com/evanphx/json-patch: open /go/src/github.com/open-cluster-management/rcm-controller/vendor/github.com/evanphx/json-patch: no such file or directory
/bin/bash: docker: command not found
make: *** [build] Error 127
2020/06/16 20:18:45 No custom metadata found and prow metadata already exists. Not updating the metadata.
2020/06/16 20:18:46 Ran for 56s
error: some steps failed:
  * could not run steps: step bin failed: could not wait for build: the build bin failed after 7s with reason DockerBuildFailed: Docker build strategy has failed.
-f build/Dockerfile
build github.com/open-cluster-management/rcm-controller/cm...or/github.com/evanphx/json-patch: no such file or directory
/bin/bash: docker: command not found
make: *** [build] Error 127
error: build error: running '/bin/bash -c set -o errexit; ...roxy.golang.org,direct" make build' failed with exit code 2
2020/06/16 20:18:46 could not load result reporting options: failed to read file "": open : no such file or directory
```

**Comment by cdoan1** [comment]
* looks like we can intiate build, but the repo has some wonkyness that has to be sorted

**Pull request closed without merging**